sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/LongLivedMessageSubscriber.java

```diff
@@ -0,0 +1,168 @@
+package com.azure.messaging.servicebus;
+
+import com.azure.core.util.logging.ClientLogger;
+import org.reactivestreams.Subscription;
+import reactor.core.Disposable;
+import reactor.core.publisher.BaseSubscriber;
+import reactor.core.publisher.Mono;
+
+import java.util.Queue;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+
+ class LongLivedMessageSubscriber extends BaseSubscriber<ServiceBusReceivedMessageContext> {
+
+    private final ClientLogger logger = new ClientLogger(SynchronousMessageSubscriber.class);
+    private final AtomicBoolean isDisposed = new AtomicBoolean(false);
+    private final long prefetch;
+    private final AtomicInteger wip = new AtomicInteger();
+
+    private volatile Subscription subscription;
+
+    private Queue<SynchronousReceiveWork> workQueue = new ConcurrentLinkedQueue<>();
+    private SynchronousReceiveWork currentWork = null;
+    private Disposable timeoutOperation;
+     private Disposable drainQueueDisposable;
+
+
+    LongLivedMessageSubscriber(long prefetch) {
+        this.prefetch = prefetch;
+    }
+
+    /**
+     * On an initial subscription, will take the first work item, and request that amount of work for it.
+     *
+     * @param subscription Subscription for upstream.
+     */
+    @Override
+    protected void hookOnSubscribe(Subscription subscription) {
+        if (this.subscription == null) {
```

Comment:
This is only subscribed to once. The null check is unnecessary. 
https://github.com/reactor/reactor-core/blob/master/reactor-core/src/main/java/reactor/core/publisher/BaseSubscriber.java#L146